test/e2e/common/runtime.go

```diff
@@ -294,7 +295,12 @@ while true; do sleep 1; done
 		}
 	}
 }`
-
+					// we might be told to use a different docker config JSON.
+					if framework.TestContext.DockerConfigFile != "" {
+						contents, err := ioutil.ReadFile(framework.TestContext.DockerConfigFile)
+						framework.ExpectNoError(err)
```

Comment:
i'd add an annotation here explaining the error in more detail:
`...err, "could not read the Docker configuration file")`